beater/beater.go

```diff
@@ -622,13 +621,46 @@ func newTransformConfig(beatInfo beat.Info, cfg *config.Config) (*transform.Conf
 	return transformConfig, nil
 }
 
-func newSourcemapStore(beatInfo beat.Info, cfg config.SourceMapping) (*sourcemap.Store, error) {
-	esClient, err := elasticsearch.NewClient(cfg.ESConfig)
+func newSourcemapStore(beatInfo beat.Info, cfg config.SourceMapping, fleetCfg *config.Fleet) (*sourcemap.Store, error) {
+	if fleetmode.Enabled() {
```

Comment:
Maybe use `if fleetCfg != nil {` here instead? We should have fleetCfg if and only if fleetmode.Enabled(), and if we don't have it then I believe the sourcemap.NewFleetStore call will panic. I guess by checking that the config is non-nil, unit testing will be a bit easier?